### DISCUSSION TITLE
More resources on Plugin Architecture

### DISCUSSION THREAD

**Discussion by AmirSolt** [fastify/fastify#5290] [General]
A few months ago I built a project in fastify and first I was skeptical on the whole plugin thing, but after using it I absolutely loved it.
I'm building another project in Go and I want to use the plugin architecture, and I have few questions.
What's the official name for it? Do you think making routes a subset of plugins a bad idea? Are there any other resources outside of fastify on this?

**Comment by mcollina** [comment]
I don't think there is an official design doc on this written, it's likely a combination of PRs and conference talks :(. It would be awesome to have somebody write it up.

**Reply by jsumners** [reply]
I'd also caution that the reasoning for the Fastify architecture is rooted in the way JavaScript and the V8 runtime work. The same concepts may not be as effective in a Go project.
